File: app/src/main/java/example/com/crackle/database/MovieDao.java
package example.com.crackle.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import example.com.crackle.model.Movie;

@Dao
public interface MovieDao {

    @Query("SELECT * FROM movie")
    LiveData<List<Movie>> getFavoritesMovies();

    @Query("SELECT is_favorite FROM movie WHERE movie_id = :movieId")
    boolean isFavorite(int movieId);

    @Query("UPDATE movie SET is_favorite = :isFavorite WHERE movie_id = :movieId" )
    void updateMovieFavorite(int movieId, boolean isFavorite);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addMovieToFavorites(Movie movie);

    @Delete
    void removeMovieFromFavorites(Movie movie);

    @Query("SELECT * FROM movie WHERE movie_id = :movieId")
    Movie getMovie(int movieId);

}

File: app/src/main/java/example/com/crackle/database/MovieDao.kt
package example.com.crackle.database

import androidx.lifecycle.LiveData
import androidx.room.*
import example.com.crackle.model.Movie

@Dao
interface MovieDao {

    @get:Query("SELECT * FROM movie")
    val favoritesMovies: LiveData<List<Movie>>

    @Query("SELECT is_favorite FROM movie WHERE movie_id = :movieId")
    fun isFavorite(movieId: Int): Boolean

    @Query("UPDATE movie SET is_favorite = :isFavorite WHERE movie_id = :movieId")
    fun updateMovieFavorite(movieId: Int, isFavorite: Boolean)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addMovieToFavorites(movie: Movie)

    @Delete
    fun removeMovieFromFavorites(movie: Movie)

    @Query("SELECT * FROM movie WHERE movie_id = :movieId")
    fun getMovie(movieId: Int): Movie

}
